fem/nonlininteg.hpp

```diff
@@ -341,6 +342,49 @@ class VectorConvectionNLFIntegrator : public NonlinearFormIntegrator
    virtual void AddMultPA(const Vector &x, Vector &y) const;
 };
 
+/** @brief This class is used to assemble the convective form of the
+nonlinear term arising in the Navier Stokes equations
+\f$(u \cdot \nabla v, w )\f$ */
+class ConvectiveVectorConvectionNLFIntegrator : public
+   VectorConvectionNLFIntegrator
+{
+private:
+   Coefficient *Q{};
+   DenseMatrix dshape, dshapex, EF, gradEF, ELV, elmat_comp;
+   Vector shape;
+public:
+   ConvectiveVectorConvectionNLFIntegrator(Coefficient &q): Q(&q) { }
+
+   ConvectiveVectorConvectionNLFIntegrator() = default;
+
+   virtual void AssembleElementGrad(const FiniteElement &el,
+                                    ElementTransformation &trans,
+                                    const Vector &elfun,
+                                    DenseMatrix &elmat);
+
+};
+/** @brief This class is used to assemble the skew-symmetric form of the nonlinear
+term arising in the Navier Stokes equations \f$.5*(u \cdot \nabla v, w ) -
+.5*(u \cdot \nabla w, v )\f$ */
```

Comment:
Besides the comment formatting, the PR looks ok.

Reply:
I believe I fixed the formatting and reran astyle. I think it looks good now.